File: app/src/main/java/com/arny/flightlogbook/adapter/FlightTypesAdapter.java
package com.arny.flightlogbook.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.arny.flightlogbook.R;
import com.arny.flightlogbook.data.models.AircraftType;
import com.arny.flightlogbook.utils.adapters.AbstractRecyclerViewAdapter;

public class FlightTypesAdapter extends AbstractRecyclerViewAdapter<AircraftType> {
    private FlightTypesListener listener;

    public FlightTypesAdapter(Context context, OnViewHolderListener listener) {
        super(context, listener);
        this.listener = (FlightTypesListener) listener;
    }

    public interface FlightTypesListener extends OnViewHolderListener {
        void onTypeEdit(int position);

        void onTypeDelete(int position);
    }

    @Override
    protected View createView(Context context, ViewGroup viewGroup, int viewType) {
        return LayoutInflater.from(getContext()).inflate(R.layout.typeitem, null);
    }

    @Override
    protected void bindView(AircraftType item, AbstractRecyclerViewAdapter.ViewHolder viewHolder) {
        if (viewHolder != null && item != null) {
            TextView typeTitle = (TextView) viewHolder.getView(R.id.typeTitle);
            ImageView edit = (ImageView) viewHolder.getView(R.id.edit);
            ImageView delete = (ImageView) viewHolder.getView(R.id.delete);
            typeTitle.setText(item.getTypeName());
            if (listener != null) {
                edit.setOnClickListener(v -> listener.onTypeEdit(viewHolder.getAdapterPosition()));
                delete.setOnClickListener(v -> listener.onTypeDelete(viewHolder.getAdapterPosition()));
            }
        }
    }
}

File: app/src/main/java/com/arny/flightlogbook/adapter/FlightTypesAdapter.kt
package com.arny.flightlogbook.adapter

import com.arny.flightlogbook.R
import com.arny.flightlogbook.data.models.AircraftType
import com.arny.flightlogbook.utils.adapters.SimpleAbstractAdapter
import kotlinx.android.synthetic.main.typeitem.view.*

class FlightTypesAdapter(private val flightTypesListener: FlightTypesListener? = null) : SimpleAbstractAdapter<AircraftType>() {

    interface FlightTypesListener : OnViewHolderListener<AircraftType> {
        fun onTypeEdit(position: Int, item: AircraftType)

        fun onTypeDelete(position: Int, item: AircraftType)
    }

    override fun getLayout(viewType: Int): Int {
        return R.layout.typeitem
    }

    override fun bindView(item: AircraftType, viewHolder: VH) {
        viewHolder.itemView.apply {
            val position = viewHolder.adapterPosition
            typeTitle.text = item.typeName
            edit.setOnClickListener {
                flightTypesListener?.onTypeEdit(position, item)
            }
            delete.setOnClickListener {
                flightTypesListener?.onTypeDelete(position, item)
            }
        }
    }

    override fun getDiffCallback(): DiffCallback<AircraftType>? {
        return object : DiffCallback<AircraftType>() {
            override fun areItemsTheSame(oldItem: AircraftType, newItem: AircraftType): Boolean {
                return oldItem.typeId == oldItem.typeId
            }

            override fun areContentsTheSame(oldItem: AircraftType, newItem: AircraftType): Boolean {
                return oldItem.typeName == oldItem.typeName
            }
        }
    }

}
